NTS/Refactor loop entity in setup boundary

StoyanGoranov opened the pull request:
Closes: https://github.com/Not-Endurance/not-timing-system/issues/210

StoyanGoranov committed:
Add Phase Entity and modify Contestant ToString

StoyanGoranov committed:
Refactor Loop and the forms bound to it

Replace Distance property in Loop with new Phase entity and apply the
necessary changes to LoopFormModel and LoopForm

StoyanGoranov committed:
Merge remote-tracking branch 'origin/dev/nts' into refactor-Loop-Entity-in-Setup-boundary

StoyanGoranov committed:
Modify Phase Entity to support separate inputs

Add Create, Update methods and appropriate ctors

https://github.com/Not-Endurance/not-timing-system/issues/210

StoyanGoranov committed:
Add ToString override to Phase

https://github.com/Not-Endurance/not-timing-system/issues/210

StoyanGoranov committed:
Modify Event Entity and EventBehind to implement new collection Phases

The list of Phases is child property of the Event object. This is done
to ensure that changes into the Phases will be reflected into the data
hierarchy.

https://github.com/Not-Endurance/not-timing-system/issues/210

StoyanGoranov committed:
Implement PhaseFormModel and its related PhaseForm and PhaseUpdate
components

https://github.com/Not-Endurance/not-timing-system/issues/210

StoyanGoranov committed:
Modify EventForm and EventFormModel to include Phases

https://github.com/Not-Endurance/not-timing-system/issues/210

StoyanGoranov committed:
Realign EventPage and use NotChildren Component for list of Official

Also adding some Mud and custom styling at NotChildren, custom styles applied
at NTS.Judge.MAUI

https://github.com/Not-Endurance/not-timing-system/issues/210

StoyanGoranov committed:
Remove OfficialList and rename OfficialPage to OfficialUpdate

OfficialList is replaced by NotChildren, OfficialUpdate renamed to
adhere to naming convention

https://github.com/Not-Endurance/not-timing-system/issues/210

StoyanGoranov committed:
Apply renamed OfficialUpdate component's route to code

This is done in a separate commit to maintain better git history

https://github.com/Not-Endurance/not-timing-system/issues/210

StoyanGoranov committed:
Add minor styling and text changes to Update components and NotChildren

Text of NotChildren rows is now vertically centered; Replacing Update
with Edit in NotChildren and Update components because from user POV
they are editing the data and then upon submission send the form to
update it at the code behind.

https://github.com/Not-Endurance/not-timing-system/issues/210

StoyanGoranov committed:
Exchange the names of the Loop and Phase entities as per spec

Only files are renamed to maintain better git history

https://github.com/Not-Endurance/not-timing-system/issues/210

StoyanGoranov committed:
Replace all Loop and Phase reference throughout the application

Applying the name change in code.

https://github.com/Not-Endurance/not-timing-system/issues/210

StoyanGoranov committed:
Restructure the Lap and Phase entities

Configured Laps can now be selected via a MultiSelect in the
configuration of Phases

https://github.com/Not-Endurance/not-timing-system/issues/210

StoyanGoranov committed:
Add interfaces, repositories and persistance for Lap

Adding ISetRepository, ReadAll method in addition to the SetRepository
and ISetState

https://github.com/Not-Endurance/not-timing-system/pull/210

StoyanGoranov committed:
Implement LapBehind and apply necessary changes to Components

These changes implement the LapBehind structure in practice

https://github.com/Not-Endurance/not-timing-system/issues/210

StoyanGoranov committed:
Ensure that changes in the set of Laps are reflected in Event hierarchy

Overrides the Update and Delete of the SetRepository with specific logic
at SetupLapRepository

https://github.com/Not-Endurance/not-timing-system/pull/221

StoyanGoranov committed:
Resolve merge conflicts with dev/nts

StoyanGoranov committed:
Resolve merge conflicts with dev/nts

achobanov commented on src-v2/NTS.Domain.Setup/Entities/Contestant.cs:
```diff
@@ -42,7 +42,7 @@ public override string ToString()
         var sb = new StringBuilder();
         var startTimeMessage = StartTimeOverride == null ? "" : "StartTime: " + $"{StartTimeOverride.Value.ToLocalTime().TimeOfDay}";
-        var IsUnrankedMessage = IsUnranked ? "Unranked: Yes" : "Unranked: No";
+        var IsUnrankedMessage = IsUnranked ? "Unranked" : "Ranked";
```
`isUnrankedMessage`

achobanov commented on src-v2/NTS.Domain.Setup/Entities/Contestant.cs:
```diff
@@ -42,7 +42,7 @@ public override string ToString()
         var startTimeMessage = StartTimeOverride == null ? "" : "StartTime: " + $"{StartTimeOverride.Value.ToLocalTime().TimeOfDay}";
-        var IsUnrankedMessage = IsUnranked ? "Unranked: Yes" : "Unranked: No";
+        var IsUnrankedMessage = IsUnranked ? "Unranked" : "Ranked";
         sb.Append($"Contestant -> Tandem: *isn't configured yet* {startTimeMessage} {IsUnrankedMessage} ");
```
If `startTimeMessage`  is empty here it would lead to duplicate spaces, right? Isn't there a better solution?

achobanov commented on src-v2/NTS.Domain.Setup/Entities/Lap.cs:
```diff
@@ -0,0 +1,36 @@
+    }
+    public double Distance { get; set; }
+
+    public string NestedToString()
+    {
+        return Distance.ToString()+"km";
+    }
```
NestedToString? Nested how? Is this ncessary?

StoyanGoranov replied:
Nested because when Lap is a property of Phase the ToString() method of Phase would need to print only the Distance property of Lap. Alternatively, the Distance property can be called directly.

achobanov replied:
That means that you are defining a member of Lap with the added context of Phase as it's parent. That's wrong. Lap doesn't know and shouldn't know how it is used. This is true for all objects, not only Domain ones.

ToString in Domain entities should be agnostic of their usage - i.e. they should provide relevant but not excessive information. In the example bellow "long" is completely unnecessary and the "Lap -" is questionable

achobanov commented on src-v2/NTS.Domain.Setup/Entities/Lap.cs:
```diff
@@ -0,0 +1,36 @@
+    }
+    public override string ToString() 
+    {
+        var phase = "Lap".Localize();
+        var sb = new StringBuilder();
+        sb.Append($"{phase} -> {Distance}km long ");
+        return sb.ToString();
```
Seems a bit overcomplicated. Do you have yo use StringBuilder here?

achobanov commented on src-v2/NTS.Domain.Setup/Entities/Loop.cs:
```diff
@@ -1,51 +1,58 @@
+    }
     [JsonConstructor]
-    public Loop(int id, double distance, int recovery, int rest) : this(distance, recovery, rest)
+    public Loop(List<Lap> laps)
```
Wait, Loop consists of Laps? Now I am entirely confused. I don't think we need all 3 of `Phase`, `Lap` and `Loop`.

achobanov commented on src-v2/NTS.Domain.Setup/Entities/Loop.cs:
```diff
@@ -1,51 +1,58 @@
+        var newLap = new Lap(distance);
+        Laps.Add(newLap);
+    }
+    public Loop(int id, double distance) 
     {
         Id = id;
+        Laps.RemoveAll(lap => lap.Id == id);
+        var updatedLap = new Lap(distance);
+        Laps.Add(updatedLap);
+    }
```
This constructor in particular seems wrong.

achobanov commented on src-v2/NTS.Domain.Setup/Entities/Phase.cs:
```diff
@@ -0,0 +1,52 @@
+        }
+        if (rest <= 0)
+        {
+            throw new DomainException(nameof(Rest), "Rest duration cannot be zero or less.");
+        }
+        SelectedLap = selectedLap;
+        Recovery = recovery;
+		Rest = rest;
+	}
+    public Lap? SelectedLap { get; private set; }
```
Do you think **Selected** adds meaningful context here?

achobanov commented on src-v2/NTS.Domain.Setup/Entities/Phase.cs:
```diff
@@ -0,0 +1,52 @@
+    public int Recovery { get; private set; }
+    public int Rest { get; private set; }
+
+    public void SetSelectedLap(Lap selectedLap)
+    {
+        SelectedLap = selectedLap;
+    }
```
Method setters like this should be avoidable. You could simply make the `SelectedLap` property with a public or internal setter. 

Furthermore, are you sure this methods works correctly? Have you  tried to update a phase after it's Lap has been set?

achobanov commented on src-v2/NTS.Domain.Setup/Entities/Phase.cs:
```diff
@@ -0,0 +1,52 @@
+        var lap = "Lap".Localize();
+        var phase = "Phase".Localize();
+		var rest = "Rest".Localize();
+        var selectedLap = SelectedLap==null? "DELETED".Localize() : SelectedLap.NestedToString();
```
Space around operators.

achobanov commented on src-v2/NTS.Persistance/Adapters/SetupLapRepository.cs:
```diff
@@ -0,0 +1,44 @@
+
+public class SetupLapRepository : SetRepository<Lap, SetupState>
+{
+    public SetupLapRepository(IStore<SetupState> store) : base(store)
```
I don't think the `Setup` prefix is necessary here unless we decide to prefix all our repositories. Currently I'm only prefixing duplicate names and there isn't a `Lap` nor `Loop` in Core.

achobanov commented on src-v2/NTS.Persistance/Adapters/SetupLapRepository.cs:
```diff
@@ -0,0 +1,44 @@
+        state = store.Load().Result;
+    }
+
+    private SetupState state { get; set; }
```
Use private fields instead of properties. By MS convention a field should only ever be private, while properties can be protected or public.

achobanov commented on src-v2/NTS.Persistance/Adapters/SetupLapRepository.cs:
```diff
@@ -0,0 +1,44 @@
+{
+    public SetupLapRepository(IStore<SetupState> store) : base(store)
+    {
+        state = store.Load().Result;
```
Perhaps I didn't emphasize enough that we don't want `Task.Result` in the codebase. At best it negates all benefits from asynchronous code, at worst it can halt the app.

Furthermore, constructors **should not** contain asynchronous logic. It's especially dangerous to call `.Result` here, because this is a **Dependency**, i.e. a service managed by the Dependency Injection Container. The container uses this constructor in ways you do not undestand and very often they are used during app startup, when configurations have not been executed yet. If this is a call to an actual database it will probably throw connectivity-related exceptions. Constructors are meant for a lightweight object initialization that can be used by the C# runtime on demand. **It's not a coincidence that they do not support async syntax**

achobanov commented on src-v2/NTS.Persistance/Adapters/SetupLapRepository.cs:
```diff
@@ -0,0 +1,44 @@
+    }
+
+    private SetupState state { get; set; }
+    public override Task<Lap> Update(Lap entity)
```
As a general rule of thumb always use `async` with methods that return Task. [A good explanation why here](https://blog.stephencleary.com/2016/12/eliding-async-await.html)

achobanov commented on src-v2/NTS.Persistance/Adapters/SetupLapRepository.cs:
```diff
@@ -0,0 +1,44 @@
+    public override Task<Lap> Update(Lap entity)
+    {
+        state.Event.Competitions
+            .Where(competition => competition.Phases.Any(phase => phase.SelectedLap.Id == entity.Id))
```
You can use `phase.SelectedLap == entity` due to `Equals` and `GetHashCode` override sin `DomainEntity.

achobanov commented on src-v2/NTS.Persistance/Adapters/SetupLapRepository.cs:
```diff
@@ -0,0 +1,44 @@
+    private SetupState state { get; set; }
+    public override Task<Lap> Update(Lap entity)
+    {
+        state.Event.Competitions
+            .Where(competition => competition.Phases.Any(phase => phase.SelectedLap.Id == entity.Id))
+            .SelectMany(competition => competition.Phases
+                .Where(phase => phase.SelectedLap.Id == entity.Id)
+                .Select(phase => new { competition, phase }))
+            .ToList()
+            .ForEach(item =>
+            {
+                item.phase.SetSelectedLap(entity);
+                item.competition.Update(item.phase);
+            });
```
Isn't
```c#
foreach (var phase in Event.Competitions.SelectMany(x => x.Phases))
{
    if (phase == entity)
    {
        phase.SetSelectedLap(entity);
    }
}
```
Simpler?

Additionally, the first `Where` clause is redundant as you are making the same check again in `SelectMany`.  Also `ToList` isn't necessary.

achobanov commented on src-v2/NTS.Persistance/Adapters/SetupLapRepository.cs:
```diff
@@ -0,0 +1,44 @@
+    }
+    public override Task<Lap> Delete(Lap entity)
+    {
+        state.Event.Competitions
+            .Where(competition => competition.Phases.Any(phase => phase.SelectedLap.Id == entity.Id))
+            .SelectMany(competition => competition.Phases
+                .Where(phase => phase.SelectedLap.Id == entity.Id)
+                .Select(phase => new { competition, phase }))
+            .ToList()
+            .ForEach(item =>
+            {
+                item.phase.SetSelectedLap(null);
+                item.competition.Update(item.phase);
+            });
```
Seems like the same code as in `Update`. Can you not reuse it?

achobanov commented on src-v2/NTS.Persistance/States/SetupState.cs:
```diff
@@ -1,14 +1,20 @@
         set => Event = value;
     }
+
+    List<Lap> ISetState<Lap>.EntitySet { 
+        get => Laps;
+    }
```
`List<Lap> ISetState<Lap>.EntitySet => Laps;` is enough

achobanov commented on src-v2/Not/Not.Application/Ports/CRUD/ISetRepository.cs:
```diff
@@ -0,0 +1,8 @@
+
+namespace Not.Application.Ports.CRUD;
+
+public interface ISetRepository<T> : ICreate<T>, IReadAll<T>, IUpdate<T>, IDelete<T>
```
Good job here. This shows a level of understanding of composition of our `IRepository` abstractions.

achobanov commented on src-v2/Not/Not.Blazor/TM/NotChildren.razor:
```diff
@@ -11,20 +11,20 @@
+        <div class="d-grid mt-1 ml-6 mb-1">
+            <div class="grid-row-1">@item</div>
+            <div class="grid-row-1 action-column">
+                <NotButtonUpdate  Size="Size.Small" OnClick="() => this._contestantNavigator.Update(UpdateRoute, item)">Edit</NotButtonUpdate>
```
Edit

achobanov commented on src-v2/Not/Not.Blazor/TM/NotSetList.razor:
This whole file seems like a lot of copy-paste. Does it have to be a separate component if they are so identical?

achobanov commented on src-v2/Not/Not.Storage/Stores/InMemoryJsonStore.cs:
```diff
@@ -5,7 +5,7 @@ namespace Not.Storage.Stores;
     where T : class, new()
 {
-    private T? _context;
+    private T _context = new();
```
Is `_context` actually used?

achobanov commented on src-v2/Not/Not.Storage/Ports/ISetState.cs:
Duplicate file caused by merge

achobanov reviewed (changes requested):
Overall pretty good, except that there are numerous build errors atm. It's a good habit to always build before pushing changes.

StoyanGoranov commented on src-v2/Judge/NTS.Judge.Blazor/Pages/Setup/Competitions/CompetitionUpdate.razor:
```diff
@@ -11,7 +11,7 @@
 
 <MudPaper Elevation="0" Class="pl-6 pr-6">
-    <NotH2>@Localizer.Get("Update Competition")</NotH2>
+    <NotH2>@Localizer.Get("Edit Competition")</NotH2>
```
Because from user POV they are about to edit the records, then after editing is completed the Update action is called which isn't performed by the user. My idea is to differentiate these and place the 'responsibility' for editing on the user and the responsibility for updating on the system.

achobanov replied:
While probably more semantically correct, I don't find it a meaningful distinction. Instead I prefer to stick to simpler scenario where we have one term and we share terminology with the user, rather then diverge from it. It would be different if from a software standpoint we need to segregate different process stages and call one Edit and another Update, but I don't think that's necessary and desirable here.

StoyanGoranov replied:
It's not that one way is more semantically correct, but from UX perspective the application should be intuitive to stimulate people to use it. UI should hint to the expected behavior.

 For this reason I haven't changed the labels of the Form buttons as the difference in software behavior is:
- Edit buttons open the *UpdateForm* 
- Update buttons send the edited data for persistence

In that sense I find that from software standpoint we already perform different actions with the different buttons and we call the entire process **Update** although it implicitly include Editing and it is up to the user to initiate the actual update or cancel the changes.

I just think having the same text will be detrimental from UX point of view, however I don't mean to argue as the final decision is yours.

achobanov replied:
I see your point, but honestly I don't think people would make that distinction. I know I didn't. Also not entirely sure it's correct: [quick google search](https://www.google.com/search?q=what%27s+the+difference+between+edit+and+update&rlz=1C1YTUH_enBG1091BG1091&oq=what%27s+the+difference+between+edit+and+update&gs_lcrp=EgZjaHJvbWUyBggAEEUYOTIHCAEQABiABDIHCAIQABiABDIHCAMQABiABDIHCAQQABiABDIHCAUQABiABDIHCAYQABiABDIHCAcQABiABDIHCAgQABiABDIHCAkQABiABNIBCDUyNDdqMGo3qAIAsAIA&sourceid=chrome&ie=UTF-8). 

I think there are better ways to spend our time. Lets keep Update everywhere and for the future lets discuss things that fall outside of the scope of the issue, before implementing them and then having to revert back later.

StoyanGoranov commented on src-v2/Judge/NTS.Judge.Blazor/Pages/Setup/Phases/PhaseForm.razor:
```diff
@@ -0,0 +1,57 @@
+@inherits NotForm<Phase>;
+
+<EditForm Model="_model">
+    <MudSelect @ref="_lapField" @bind-Value="_model.SelectedLap" Label="@Localizer.Get("Phase Distance")" @onchange="(_) => SubmitUpdate()"
```
Seems like a good idea since MudSelect is already used 3 times but there is a problem with the implementation since each MudSelect may represent a selectable list of different object types.

However the NotSelect must accept a generic type parameter:
`@typeparam T where T : DomainEntity`

An issue arises immediately at the `OfficialForm.razor` component because it uses and Enum to provide the items of the Select.

StoyanGoranov committed:
address review

https://github.com/Not-Endurance/not-timing-system/pull/221

StoyanGoranov committed:
Merge branch 'refactor-Loop-Entity-in-Setup-boundary' of https://github.com/Not-Endurance/not-timing-system into refactor-Loop-Entity-in-Setup-boundary

StoyanGoranov committed:
rename SetupLapRepository to LapRepository

https://github.com/Not-Endurance/not-timing-system/issues/210

achobanov commented:
Also a reminder for the branch name convention - nts/<feature> or dev/<feature> (if working on the legacy app)

StoyanGoranov committed:
fix Person serialization bug

ensure the Person ctor marked with JsonConstructor performs correct
deserialization of objects that have been persisted

StoyanGoranov committed:
Implement NotSelect component and replace MudSelect

Create a NotSelect component that uses a NotListModel.cs. The model can
convert Enums and DomainEntity types to a IEnumerables consumable by the
NotSelect. The new component is used at CompetitionForm, OfficialForm
and PhaseForm.

StoyanGoranov committed:
Use INotSetBehind interface in EventBehind and CompetitionChildrenBehind

This will ensure that a single NotSetList component can be used to list
entities instead of juggling between NotSetList and NotChildren

https://github.com/Not-Endurance/not-timing-system/pull/221

StoyanGoranov committed:
Change ToString implementation of Lap and Phase, Delete Loop entity

Simplify ToString logic. Loop entity is redundant. Lap will be renamed
Loop in future commit.

https://github.com/Not-Endurance/not-timing-system/pull/221

StoyanGoranov committed:
Rename Lap to Loop

https://github.com/Not-Endurance/not-timing-system/pull/221

StoyanGoranov committed:
Change file names containing Lap to Loop

https://github.com/Not-Endurance/not-timing-system/pull/221

StoyanGoranov committed:
Revert 'Edit' to 'Update' in forms and remove unused ISetRepository

https://github.com/Not-Endurance/not-timing-system/pull/221

achobanov commented on src-v2/Judge/NTS.Judge.Blazor/Pages/Setup/Competitions/CompetitionForm.razor:
```diff
@@ -45,7 +42,7 @@
         RegisterInjector(nameof(_model.Name), () => _nameField);
         RegisterInjector(nameof(_model.StartTime), () => _timeField);
-        RegisterInjector(nameof(_model.Type), () => _typeField);
+        //RegisterInjector(nameof(_model.Type), () => _typeField);
```
Is there a problem with the validation?

StoyanGoranov replied:
Currently, the RegisterInjector only accepts `Func<MudBaseInput<TInput>>` or `Func<MudPicker<TInput>>` as a second argument. We need to extend it to also support Custom components because the _typeField is no longer of type `MudSelect<CompetitionType>` but `NotSelect<CompetitionType>` instead.

I have left a comment:
// TODO: Implement Validation injection for custom input components
at line 17 NotSelect.razor to ensure the needed improvements aren't forgotten.

achobanov replied:
Makes sense. It's probably going to stay like that for the demo version and then for NTS 1 we'll probably end up reworking the validation anyway.

achobanov commented on src-v2/Judge/NTS.Judge.Blazor/Pages/Setup/Officials/OfficialForm.razor:
```diff
@@ -4,29 +4,35 @@
 
+    private IEnumerable<NotListModel<OfficialRole>> _items = NotListModel.FromEnum<OfficialRole>();
+
+    protected override void OnInitialized()
+    {
+        base.OnInitialized();
+    }
```
Seems unnecessary

achobanov commented on src-v2/Judge/NTS.Judge.Blazor/Pages/Setup/Officials/OfficialForm.razor:
```diff
@@ -4,29 +4,35 @@
         RegisterInjector(nameof(_model.Name), () => _nameField);
-        RegisterInjector(nameof(_model.Role), () => _roleField);
-    }
+        // RegisterInjector(nameof(_model.Role), () => _roleField);
```
Validation?

achobanov commented on src-v2/Judge/NTS.Judge.Blazor/Pages/Setup/Phases/PhaseForm.razor:
```diff
@@ -1,41 +1,40 @@
     private PhaseFormModel _model { get; set; } = new();
-    private IEnumerable<Lap> Items { get; set; } = default!;
+
+    private IEnumerable<NotListModel<Loop>> _items = default!;
```
Isn't `_items` a bit generic? What happens if we add a second `NotSelect` for example?

achobanov commented on src-v2/Judge/NTS.Judge.Blazor/Pages/Setup/Phases/PhaseForm.razor:
```diff
@@ -1,41 +1,40 @@
     public override void RegisterValidationInjectors()
     {
-        RegisterInjector(nameof(_model.SelectedLap), () => _lapField);
+        //RegisterInjector(nameof(_model.Loop), () => _loopField);
```
Validation

achobanov commented on src-v2/Judge/NTS.Judge/Events/CompetitionChildrenBehind.cs:
```diff
@@ -2,9 +2,10 @@
 
 namespace NTS.Judge.Events;
-public class CompetitionChildrenBehind : INotBehindParent<Contestant>, INotBehindParent<Phase>, INotBehindWithChildren<Competition>
+public class CompetitionChildrenBehind : INotSetBehind<Contestant>, INotSetBehind<Phase>, INotBehindWithChildren<Competition>
```
This might be a good place to rename the odd `INotBehindWithChildren` to a more reasonable `INotBehindParent` since  we no longer have another interface conflicting.

StoyanGoranov replied:
Renamed the interface, however used the name `INotParentBehind` to avoid the need to remove the `INotBehindParent` yet. And also comply with the emerging naming convention -> INot-*semantic part of the name*-Behind. 

Looking forward to hearing what you think about that approach.

achobanov replied:
Sounds great, I like consistency. However why not delete the old `INotBehindParent`?

StoyanGoranov replied:
Done.

achobanov commented on src-v2/NTS.Domain.Setup/Entities/Loop.cs:
```diff
@@ -1,58 +1,30 @@
     {
-        Laps.RemoveAll(lap => lap.Id == child.Id);
-        Laps.Add(child);
+        var message = $"{Distance}km";
+        return message;
```
Can be single line. Also km isn't localized.

achobanov commented on src-v2/NTS.Domain.Setup/Entities/Phase.cs:
```diff
@@ -4,19 +4,19 @@
 
     [JsonConstructor]
-    public Phase(int id, Lap selectedLap, int recovery, int rest)
+    public Phase(int id, Loop selectedLoop, int recovery, int rest)
```
Isn't this `selectedLoop`  name a problem? Does `selected` bring meaningful context?

achobanov commented on src-v2/NTS.Persistance/Adapters/LoopRepository.cs:
```diff
@@ -3,54 +3,54 @@
     }
 
-    public async Task UpdateLapValue(Lap entity, bool isDelete=false) 
+    public async Task UpdateLoopValue(Loop entity, bool isDelete=false) 
```
Does it have to be public?

StoyanGoranov committed:
address review 16.6.2024

https://github.com/Not-Endurance/not-timing-system/pull/221

achobanov reviewed (approved):
Good job on renaming `_items` in multiple places, not only where I had left a comment

StoyanGoranov committed:
remove INotBehindParent interface and NotChildren component

https://github.com/Not-Endurance/not-timing-system/pull/221
